libcaf_test/caf/test/runnable.hpp

```diff
@@ -173,20 +215,52 @@ class CAF_TEST_EXPORT runnable {
       reporter::instance->fail("throws Exception", location);
     }
   }
-#endif
 
-  template <class Expr>
-  void should_fail(Expr&& expr, const caf::detail::source_location& location
-                                = caf::detail::source_location::current()) {
+  /// Checks whether `expr()` throws an exception of type `Exception` and
+  /// increases the failure count.
+  template <class Exception = void, class Expr>
+  void should_fail_with_exception(Expr&& expr,
+                                  const caf::detail::source_location& location
+                                  = caf::detail::source_location::current()) {
     auto* rep = reporter::instance;
     auto before = rep->test_stats();
-    expr();
+    auto lvl = rep->verbosity(CAF_LOG_LEVEL_QUIET);
+    auto caught = false;
+    if constexpr (std::is_same_v<Exception, void>) {
+      try {
+        expr();
+      } catch (...) {
+        caught = true;
+      }
+    } else {
+      try {
+        expr();
+      } catch (const Exception&) {
+        caught = true;
+      } catch (...) {
+        // TODO: print error message
```

Comment:
Can you resolve this TODO?

Reply:
No, the function doesn't exist yet. Needs the printing functions from https://github.com/actor-framework/actor-framework/issues/1479.